src/main/cli/cli.c

```diff
@@ -1315,30 +1316,46 @@ static void cliSerial(const char *cmdName, char *cmdline)
 
     uint8_t validArgumentCount = 0;
 
-    const char *ptr = cmdline;
+    char *ptr = cmdline;
+    char *tok = strsep(&ptr, " ");
+    serialPortIdentifier_e identifier = findSerialPortByName(tok, strcasecmp);
+    if (identifier == SERIAL_PORT_NONE) {
+        char *eptr;
+        identifier = strtoul(tok, &eptr, 10);
+        if (*eptr) {
+            // parsing ended before end of token indicating an invalid identifier
+            identifier = SERIAL_PORT_NONE;
+        } else {
+            // correction for legacy configuration where UART1 == 0
+            if (identifier >= SERIAL_PORT_LEGACY_START_IDENTIFIER && identifier < SERIAL_PORT_START_IDENTIFIER) {
+                identifier += SERIAL_PORT_UART1;
+            }
+       }
+    }
 
-    int val = atoi(ptr++);
-    serialPortConfig_t *currentConfig = serialFindPortConfigurationMutable(val);
+    serialPortConfig_t *currentConfig = serialFindPortConfigurationMutable(identifier);
 
-    if (currentConfig) {
-        portConfig.identifier = val;
-        validArgumentCount++;
+    if (!currentConfig) {
+        cliShowParseError(cmdName);
+        return;
     }
```

Comment:
This seems to resolve restoring the diff
```suggestion
    if (currentConfig) {
        portConfig = *currentConfig;
    } else {
        cliShowParseError(cmdName);
        return;
    }
```